packages/common/src/internal/clients/cache/AbstractCacheClient.ts

```diff
@@ -44,4 +61,1062 @@ export abstract class AbstractCacheClient implements ICacheClient {
   public async listCaches(): Promise<ListCaches.Response> {
     return await this.controlClient.listCaches();
   }
+
+  /**
+   * Gets the value stored for the given key.
+   *
+   * @param {string} cacheName - The cache to perform the lookup in.
+   * @param {string | Uint8Array} key - The key to look up.
+   * @returns {Promise<CacheGet.Response>} -
+   * {@link CacheGet.Hit} containing the value if one is found.
+   * {@link CacheGet.Miss} if the key does not exist.
+   * {@link CacheGet.Error} on failure.
+   */
+  public async get(
+    cacheName: string,
+    key: string | Uint8Array
+  ): Promise<CacheGet.Response> {
+    return await this.getNextDataClient().get(cacheName, key);
+  }
+
+  /**
+   * Associates the given key with the given value. If a value for the key is
+   * already present it is replaced with the new value.
+   *
+   * @param {string} cacheName - The cache to store the value in.
+   * @param {string | Uint8Array} key - The key to set.
+   * @param {string | Uint8Array} value - The value to be stored.
+   * @param {SetOptions} [options]
+   * @param {number} [options.ttl] - The time to live for the item in the cache.
+   * Uses the client's default TTL if this is not supplied.
+   * @returns {Promise<CacheSet.Response>} -
+   * {@link CacheSet.Success} on success.
+   * {@link CacheSet.Error} on failure.
+   */
+  public async set(
+    cacheName: string,
+    key: string | Uint8Array,
+    value: string | Uint8Array,
+    options?: SetOptions
+  ): Promise<CacheSet.Response> {
+    const client = this.getNextDataClient();
+    return await client.set(cacheName, key, value, options?.ttl);
+  }
+
+  /**
+   * Removes the given key from the cache. The key can represent a single value
+   * or a collection.
+   *
+   * @param {string} cacheName - The cache to delete from.
+   * @param {string | Uint8Array} key - The key to delete.
+   * @returns {Promise<CacheDelete.Response>} -
+   * {@link CacheDelete.Success} on success.
+   * {@link CacheDelete.Error} on failure.
+   */
+  // public async delete(
```

Comment:
I copied all the signatures over from our nodejs client, and just commented out the ones that we havent implemented yet. We will implement these soon so felt like it was fine to leave here for now